code/clue.js

```diff
@@ -30,35 +129,145 @@ const suspects = [
 const pickKiller = () => {
   // This will randomly select a killer from the suspects. And add that to the mystery object.
   mystery.killer = randomSelector(suspects)
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer.
+  // This will change the card to the one connected to the chosen killer id.
   const theKiller = document.getElementById("killer")
   const theKillerName = document.getElementById("killerName")
+  const theKillerAge = document.getElementById("killerAge")
+  const theKillerOccup = document.getElementById("killerOccupation")
+  const theKillerDescript = document.getElementById("killerDescription")
+  const theKillerImage = document.getElementById("killerImage")
+
 
   theKiller.style.background = mystery.killer.color
-  theKillerName.innerHTML =
-    mystery.killer.firstName + " " + mystery.killer.lastName
+  theKillerName.innerHTML = mystery.killer.firstName + " " + mystery.killer.lastName
+  theKillerAge.innerHTML = mystery.killer.age
+  theKillerOccup.innerHTML = mystery.killer.occupation
+  theKillerDescript.innerHTML = mystery.killer.description
+  theKillerImage.src =  mystery.killer.image
+
 }
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+// This function will be invoked when you click on the weapon card.
+const pickWeapon = () => {
+  mystery.weapon = randomSelector(weapons)
+
+  // This will change the card to the one connected to the chosen weapon id.
+  const theWeapon = document.getElementById("weapon")
+  const theWeaponName = document.getElementById("weaponName")
+  const theWepaonWeight = document.getElementById("weaponWeight")
+  const theWeaponImage = document.getElementById("weaponImage")
+ 
+
+  theWeaponName.innerHTML = mystery.weapon.name
+  theWepaonWeight.innerHTML = mystery.weapon.weight
+  theWeaponImage.src = mystery.weapon.image
+  theWeapon.style.background = "white";
+
+}
+
+// This function will be invoked when you click on the weapon card.
+const pickRoom = () => {
+  mystery.room = randomSelector(rooms)
+
+  // This will change the card to the one connected to the chosen room id.
+  const theRoom = document.getElementById("room")
+  const theRoomName = document.getElementById("roomName")
+
+  theRoom.style.background = "gold"
+  theRoomName.innerHTML = mystery.room
+}
+
+// This function will pick the mystery when clicking on "pick mystery" button, instead of clicking the cards. 
+// Only visible in phone size. 
+const pickMystery = () => {
+
+  mystery.killer = randomSelector(suspects)
+  mystery.weapon = randomSelector(weapons)
+  mystery.room = randomSelector(rooms)
+  
+  document.getElementById("pickAllCard").style.display ="none"
+}
+
+
+
+// This function will reveal the mystery and replace the cards with a text of the murder, weapon and room when clicking the button. 
+
+const revealMystery = () => {
+  if (mystery.killer || mystery.weapon || mystery.room === onclick) {
```

Comment:
When you have `||` it means "or". So in this case if I only click on killer, it goes trough here (because it's `true`) and gives the used a sentence with `undefined` in it, which is not so nice. Either you add a if statement for the result (if the user can choose only one card) or you can change `||` to `&&`. `&&` means "and".